src/main/java/br/com/academiadev/thunderpets/controller/UsuarioController.java

```diff
@@ -0,0 +1,127 @@
+    public PageImpl<UsuarioDTO> listar(@RequestParam(defaultValue = "0") int paginaAtual,
+                                       @RequestParam(defaultValue = "10") int tamanho,
+                                       @RequestParam(defaultValue = "ASC") Sort.Direction direcao,
+                                       @RequestParam(defaultValue = "nome") String campoOrdenacao) {
+        PageRequest paginacao = PageRequest.of(paginaAtual, tamanho, direcao, campoOrdenacao);
+        Page<Usuario> paginaUsuarios = usuarioRepository.findAll(paginacao);
+        int totalDeElementos = (int) paginaUsuarios.getTotalElements();
+
+        return new PageImpl<UsuarioDTO>(paginaUsuarios.stream()
+                .map(usuario -> converterUsuarioParaUsuarioDTO(usuario)).collect(Collectors.toList()),
+                paginacao,
+                totalDeElementos);
+    }
+
+    @GetMapping("/{id}")
+    public ResponseEntity<UsuarioDTO> buscar(@PathVariable("id") UUID id) {
+        return ResponseEntity.ok(converterUsuarioParaUsuarioDTO(usuarioRepository.findById(id).get()));
+    }
+
+    @PostMapping("/")
+    public ResponseEntity<Object> salvar(@RequestBody UsuarioDTO usuarioDTO) {
+        UsuarioDTO usuarioPersistido = new UsuarioDTO();
+        try {
+            Usuario usuario = usuarioDTO.getUsuario();
+            usuario = usuarioRepository.saveAndFlush(usuario);
+
+            List<Contato> contatosDoUsuario = contatoRepository.findByUsuario(usuario);
+            for (Contato contatoDelete : contatosDoUsuario) {
+                contatoRepository.delete(contatoDelete);
+            }
+
+            for (ContatoDTO contatoDTO : usuarioDTO.getContatos()) {
+                Contato contato = new Contato();
+                contato.setId(contatoDTO.getId());
+                contato.setTipo(contatoDTO.getTipo());
+                contato.setDescricao(contatoDTO.getDescricao());
+                contato.setUsuario(usuarioDTO.getUsuario());
+                contatoRepository.saveAndFlush(contato);
+            }
+
+            usuarioPersistido = converterUsuarioParaUsuarioDTO(usuario);
+        } catch (Exception e) {
+            return ResponseEntity.status(500).body(e.getMessage());
+        }
+
+        return ResponseEntity.ok(usuarioPersistido);
+    }
+
+    @DeleteMapping("/{id}")
+    public ResponseEntity<Object> deletar(@PathVariable("id") UUID id) {
+        try {
+            Usuario usuario = usuarioRepository.findById(id).get();
+            usuario.setAtivo(false);
+            usuarioRepository.saveAndFlush(usuario);
+        } catch (Exception e) {
+            return ResponseEntity.status(500).body(e.getMessage());
+        }
+
+        return ResponseEntity.ok(true);
+    }
+
+    @GetMapping("{id}/foto")
+    public ResponseEntity<Object> getFoto(@PathVariable("id") UUID id) {
+        byte[] bytes = null;
+        try {
+            Usuario usuario = usuarioRepository.findById(id).get();
+            bytes = usuario.getFoto();
```

Comment:
Se a foto for nula, precisamos explicitar isso pro front. Talvez retornar um outro código, para que eles possam usar a imagem padrão.